src/countCakes.test.js

```diff
@@ -8,13 +8,97 @@ describe(`Function 'countCakes':`, () => {
   });
 
   it(`should return a number`, () => {
+    expect(typeof countCakes()).toBe('number');
+  });
+
+  it(`should return the maximum number of cakes that
+  Pete can bake according to the recipe`, () => {
+    const rcp = {
+      flour: 500,
+      sugar: 200,
+      eggs: 1,
+    };
 
+    const ing = {
+      flour: 1200,
+      sugar: 1200,
+      eggs: 5,
+      milk: 200,
+    };
+
+    expect(countCakes(rcp, ing)).toBe(2);
   });
 
-  it(`should return the number of cakes 
-    when Pete has as many ingredients as the recipe requires`, () => {
+  it(`should return 1 when Pete has the same number
+  of ingredients as the recipe requires`, () => {
+    const rcp = {
+      flour: 500,
+      sugar: 200,
+      eggs: 1,
+    };
+
+    const ing = {
+      flour: 500,
+      sugar: 200,
+      eggs: 1,
+    };
 
+    expect(countCakes(rcp, ing)).toBe(1);
   });
 
-  // write your tests here
+  it(`should return 0 when Pete has
+  not enough ingredients as the recipe requires`, () => {
+    const rcp = {
+      apples: 3,
+      flour: 300,
+      sugar: 150,
+      milk: 100,
+      oil: 100,
+    };
+
+    const ing = {
+      sugar: 500,
+      flour: 2000,
+      milk: 2000,
+    };
+
+    expect(countCakes(rcp, ing)).toBe(0);
+  });
+
+  it(`should consider the ingredients that are not present as 0`, () => {
+    const rcp = {
+      flour: 300,
+      sugar: 150,
+      milk: 100,
+      oil: 100,
+    };
+    const ing = {
+      sugar: 500,
+      flour: 2000,
+      milk: 2000,
+    };
+
+    expect(countCakes(rcp, ing)).toBe(0);
+  });
+
+  it(`should return the type 'number' for ingredient values`, () => {
+    const rcp = {
+      flour: 300,
+      sugar: 150,
+      milk: 100,
+      oil: 100,
+    };
+    const ing = {
+      apples: 3,
+      sugar: 500,
+      flour: 2000,
+      milk: 2000,
+    };
+
+    for (const i in rcp) {
+      for (const j in rcp) {
+        expect(typeof countCakes(rcp[i], ing[j])).toBe('number');
```

Comment:
This test is not needed, you should check only the result of the function, not its input